### PR TITLE
Pr/create new analytics view

### PR DESCRIPTION
**Pull request by andgonzalezcas**
"Updated analytics charts component:

- Added enums and interfaces for better data organization.
- Refactored code to separate functional view and UI components.
- Utilized 'react_redux' for managing application currencies.
- Improved folder structure to enhance code organization.

These changes aim to enhance the analytics charts functionality, improve maintainability, and promote code reusability. The updates allow for easier management of currencies, better separation of concerns, and more straightforward navigation of the project structure.

### CODE REVIEW

**Comment by nanSer0** [comment]
Fix styles from figma

**Review by nanSer0 — CHANGES_REQUESTED** [review]
update styles from figma

File: src/app/layout.tsx [hunk 1]
```diff
@@ -8,33 +8,37 @@ import NavbarUI from '../ui/navbar/navbar.ui';
...
 import SubHeader from '../ui/subHeader/subHeader.ui';
+import { Provider } from 'react-redux';
+import store from '../state/store';
+import { fetchCurrencyData } from '../utils/api';
+import { setCurrencyValues } from '../state/currencySlice';
 
-const plusJakarta = Plus_Jakarta_Sans({ subsets: ['latin'], display: 'block', variable: '--jakarta-font'});
-const fontIcons = FontIcons({src: '../../public/fonts/fonts-icons/iconSet01.ttf', display: 'block', variable: '--icons-font'})
+const plusJakarta = Plus_Jakarta_Sans({ subsets: ['latin'], display: 'block', variable: '--jakarta-font' });
+const fontIcons = FontIcons({ src: '../../public/fonts/fonts-icons/iconSet01.ttf', display: 'block', variable: '--icons-font' })
```
Comment (nanSer0) [inline src/app/layout.tsx]: add semicolon

File: src/app/layout.tsx [hunk 2]
```diff
@@ -64,33 +68,45 @@ export default function RootLayout({
   const [banner, setBanner] = useState<boolean>(false);
   const pathname = usePathname()
```
Comment (nanSer0) [inline src/app/layout.tsx]: add semicolon

File: src/app/layout.tsx [hunk 3]
```diff
@@ -64,33 +68,45 @@ export default function RootLayout({
...
+    if (isConnected) {
+      if (pathname !== '/metaverseexplorer/watchlist' && pathname !== '/metaverseexplorer/analytics') {
         setBanner(true);
-      }else{
+      } else {
         setBanner(false)
       }
     }
-  },[pathname, isConnected])
+  }, [pathname, isConnected])
```
Comment (nanSer0) [inline src/app/layout.tsx]: we should review it with @smarquezmgh

File: src/app/layout.tsx [hunk 4]
```diff
@@ -64,33 +68,45 @@ export default function RootLayout({
...
+
+  useEffect(() => {
+    // Asynchronous function that updates data for Currency coins
+    const updateCurrencyData = async () => {
+      const currencyData = await fetchCurrencyData();
+      if (currencyData) { store.dispatch(setCurrencyValues(currencyData)); }
```
Comment (nanSer0) [inline src/app/layout.tsx]: remove '{' and '}'

File: src/state/store.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { configureStore } from '@reduxjs/toolkit'
```
Comment (nanSer0) [inline src/state/store.ts]: add semicolon

File: src/ui/analytics/chart/chartGridSelector.ui.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      <button
+        onClick={() => { setGridChartCols('col-span-1') }}
+        className={`grid rounded-lg w-12 h-12 place-content-center justify-items-center bg-nm-fill 
+        ${(gridChartCols === 'col-span-1') ? 'shadow-inset' : 'shadow-relief-12'}`}
+      >
+        <Image
+          src='/images/icons/chart/ic_two-col.svg' // Path to the image file
+          width={22} // Width of the image
+          height={22} // Height of the image
+          alt="button to change charts grid to 2 cols" // Alternate text for accessibility
```
Comment (nanSer0) [inline src/ui/analytics/chart/chartGridSelector.ui.tsx]: maybe we don't need this deep comments!

File: src/ui/analytics/chart/chartNameFilterSelector.ui.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        className="cursor-pointer"
+        onChange={(e) => {
+          e.stopPropagation();
+          setChecked(!checked);
+          handleChange();
+        }}
+      />
+    </button>
+  );
+};
```
Comment (nanSer0) [inline src/ui/analytics/chart/chartNameFilterSelector.ui.tsx]: Create a new file for it

File: src/ui/analytics/chart/linearChart.ui.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  chartApiData: AnalyticChartData; // Data for the linear chart
+  chartOptions: {
+    cols: 'col-span-1' | 'col-span-2'; // Column span options for chart layout
+    visibleDays: number; // Number of visible days for the chart
+    visibleMetaverses: Metaverses[]; // Array of visible metaverses for the chart
+    currencySelected: Currencies; // Selected currency for the chart
+  };
+}
+
+//* Main component for the linear chart.
```
Comment (nanSer0) [inline src/ui/analytics/chart/linearChart.ui.tsx]: this comment doesn't add util info to component (maybe you want mark the main component, it is not necessary becouse you should have the components on differents files)

File: src/ui/analytics/chart/linearChart.ui.tsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    }
+
+    LinearChart.updateSymbol(chartOptions.currencySelected);
+    LinearChart.ChangeVisibleTimelapsRange(chartOptions.visibleDays);
+    LinearChart.createTooltip(chartElement);
+
+    // Cleanup function to remove the chart when the component unmounts
+    return () => { LinearChart.remove(); };
+  }, [chartOptions, currencyData]);
```
Comment (nanSer0) [inline src/ui/analytics/chart/linearChart.ui.tsx]: we can review it deeply

**Review by nanSer0 — CHANGES_REQUESTED** [review]

File: src/ui/analytics/chart/filterNameOption.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import { useState } from "react";
```
Comment (nanSer0) [inline src/ui/analytics/chart/filterNameOption.tsx]: this component doesn't have the correct name, it should be named ........   .ui.tsx

**Review by nanSer0 — CHANGES_REQUESTED** [review]

**Review by nanSer0 — APPROVED** [review]

**Comment by OswaldUnity** [comment]
Please fix conflict first

**Pull request merged**